xarray/core/dataarray.py

```diff
@@ -2099,6 +2136,14 @@ def stack(
             replace. An ellipsis (`...`) will be replaced by all unlisted dimensions.
             Passing a list containing an ellipsis (`stacked_dim=[...]`) will stack over
             all dimensions.
+        create_index : bool, optional
+            If True (default), create a multi-index for each of the stacked dimensions.
+            If False, don't create any index.
+            If None, create a multi-index only if exactly one single (1-d) coordinate
+            index is found for every dimension to stack.
+        index_cls: class, optional
```

Comment:
Could we address this later, e.g., when making indexes (i.e., `Index`, `PandasIndex` and `PandasMultiIndex`) part of the public API? We can also remove it until then (although it would prevent hacking with custom indexes).

Reply:
of course! This is not public API yet, so this will not be rendered for now. We just have to make sure we don't forget.

Reply:
> We just have to make sure we don't forget.

I updated #6293.